File: Assignment 3/Assn3.cpp
#include <string>
#include <fstream>
#include <regex>
#include <iostream>

using namespace std;

string dec_digit ("[0-9]");
string hex_digit (dec_digit + "|[a-f]|[A-F]");
string exponent ("p|P");
string suff ("(f|F|l|L)?");
string sign ("(\\+|-)?");
string hexa ("(" + hex_digit + ")+"); 
string hex_floating_point ("^(0x|0X)(" + hexa + "|" + hexa + "." + hexa + "|." + hexa + ")(" + exponent + ")(" + sign + ")(" + dec_digit +")(" + suff + ")$");

regex re(hex_floating_point);

int main(int argc, char **argv){
    ifstream fs;
    string line;

    fs.open(argv[1]);

    //cout << hex_floating_point << endl;
    while(getline(fs,line)){
        if (regex_match(line, re))
            cout << "Matched: " << line << endl;
        else    
            cout << "Not Matched: "<< line << endl;
    }

    return 0;
}

File: Assignment 3/Assn3.py
import re
import sys

dec_digit = '[0-9]'
hex_digit = f'{dec_digit}|[a-f]|[A-F]'
exp = 'p|P'
suff = '(f|F|l|L)?' 
sign = '(\\+|-)?'
hexa = f'({hex_digit})+' 
hex_floating_point = f'^(0x|0X)({hexa}|{hexa}.{hexa}|.{hexa})({exp})({sign})({dec_digit}+)({suff})$'

reg = re.compile(hex_floating_point)

filepath = sys.argv[1]

fp = open(filepath)
#print (f'Regular expression for a hexadecimal floating point: {hex_floating_point}')
for line in fp.readlines():
    line = line.strip()
    print(f'{"Matched:" if reg.search(line) else "Not Matched:"} {line}')